### PR TITLE
Add --replace flag to "podman container runlabel"

### PR DESCRIPTION
**Pull request by jwhonce**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1677908

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Did @baude beat you to it with #2594?

**Comment by TomSweeneyRedHat** [comment]
dupe of #2594?

**Comment by jwhonce** [comment]
@TomSweeneyRedHat I think @baude forgot I was working this, but solutions are different enough to discuss.  We'll close one of these PR's.

File: docs/podman-container-runlabel.1.md [hunk 1]
```diff
@@ -97,6 +98,10 @@ Require HTTPS and verify certificates when contacting registries (default: true)
 then TLS verification will be used. If set to false, then TLS verification will not be used. If not specified,
 TLS verification will be used unless the target registry is listed as an insecure registry in registries.conf
 
+**--replace**
```
Comment (baude) [inline docs/podman-container-runlabel.1.md]: need to to bash completions

File: docs/podman-container-runlabel.1.md [hunk 2]
```diff
@@ -97,6 +98,10 @@ Require HTTPS and verify certificates when contacting registries (default: true)
...
+
+If a container exists of the default or given name, it will be replaced with a container from this image
```
Comment (baude) [inline docs/podman-container-runlabel.1.md]: I think you should be more explicit here.  Replace means you will force delete the previous container (even if it is running) and then make a new one.

File: cmd/podman/runlabel.go [hunk 3]
```diff
@@ -146,10 +147,26 @@ func runlabelCmd(c *cliconfig.RunlabelValues) error {
...
-		fmt.Printf("Command: %s\n", strings.Join(cmd, " "))
+		fmt.Printf("command: %s\n", strings.Join(cmd, " "))
 		if c.Display {
 			return nil
 		}
 	}
+
+	// If container already exists && --replace given -- Nuke it
+	if c.Replace {
+		for i, entry := range cmd {
```
Comment (baude) [inline cmd/podman/runlabel.go]: food for thought, would shlex.split and index func avoid some of this nesting?  again, just a thought

File: cmd/podman/cliconfig/config.go [hunk 4]
```diff
@@ -416,6 +416,7 @@ type RunlabelValues struct {
 	Quiet           bool
 	SignaturePolicy string
 	TlsVerify       bool
+	Replace         bool
```
Comment (baude) [inline cmd/podman/cliconfig/config.go]: i try to keep this in alpha sorted order

File: cmd/podman/runlabel.go (hunk 3 above)
Comment (baude) [reply cmd/podman/runlabel.go]: ok it is already split ...

File: cmd/podman/runlabel.go [hunk 5]
```diff
@@ -146,10 +147,26 @@ func runlabelCmd(c *cliconfig.RunlabelValues) error {
...
 	if !c.Quiet {
-		fmt.Printf("Command: %s\n", strings.Join(cmd, " "))
+		fmt.Printf("command: %s\n", strings.Join(cmd, " "))
 		if c.Display {
 			return nil
 		}
 	}
+
+	// If container already exists && --replace given -- Nuke it
+	if c.Replace {
```
Comment (baude) [inline cmd/podman/runlabel.go]: wwyt about if c.Replace && util.StringInSlice(cmd, "--name") {

**Comment by baude** [comment]
i will withdraw my PR and support this one with the recommended changes. some food for thought, others not so much.

File: cmd/podman/cliconfig/config.go (hunk 4 above)
Comment (jwhonce) [reply cmd/podman/cliconfig/config.go]: done

File: cmd/podman/runlabel.go (hunk 5 above)
Comment (mheon) [reply cmd/podman/runlabel.go]: Should we also verify that cmd[0] is podman (or /proc/self/bin or similar)?

File: cmd/podman/runlabel.go [hunk 6]
```diff
@@ -146,10 +147,26 @@ func runlabelCmd(c *cliconfig.RunlabelValues) error {
...
+		for i, entry := range cmd {
+			if entry == "--name" {
+				ctr, err := runtime.LookupContainer(cmd[i+1])
+				if err == nil {
```
Comment (mheon) [inline cmd/podman/runlabel.go]: `if err != nil && errors.Cause(err) != libpod.ErrNoSuchCtr` should at the very least log the error

File: cmd/podman/runlabel.go (hunk 5 above)
Comment (jwhonce) [reply cmd/podman/runlabel.go]: util.StringInSlice doesn't give me the index. I would need to iterate the slice twice if I used that.

File: cmd/podman/runlabel.go [hunk 7]
```diff
@@ -146,10 +149,31 @@ func runlabelCmd(c *cliconfig.RunlabelValues) error {
...
 		if c.Display {
 			return nil
 		}
 	}
+
+	// If container already exists && --replace given -- Nuke it
+	if c.Replace {
+		for i, entry := range cmd {
+			if entry == "--name" {
+				ctr, err := runtime.LookupContainer(cmd[i+1])
```
Comment (rhatdan) [inline cmd/podman/runlabel.go]: Does this work correctly if cmd[i+1] is an environment variable like $NAME or $IMAGE?

**Comment by mheon** [comment]
LGTM
@baude @rhatdan PTAL

File: cmd/podman/runlabel.go [hunk 8]
```diff
@@ -45,6 +47,7 @@ func init() {
 	flags.StringVar(&runlabelCommand.CertDir, "cert-dir", "", "`Pathname` of a directory containing TLS certificates and keys")
 	flags.StringVar(&runlabelCommand.Creds, "creds", "", "`Credentials` (USERNAME:PASSWORD) to use for authenticating to a registry")
 	flags.BoolVar(&runlabelCommand.Display, "display", false, "Preview the command that the label would run")
+	flags.BoolVar(&runlabelCommand.Replace, "replace", false, "Replace existing container with new one from image")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/runlabel.go]: perhaps:  "with a new one from the image"

**Comment by TomSweeneyRedHat** [comment]
LGTM, one nit/suggestion for wording.

**Pull request merged**